integration_tests/test_simple_cross_evaluation.py

```diff
@@ -28,60 +28,61 @@ async def simple_cross_evaluation(n_battles, format_):
     await cross_evaluate(players, n_challenges=n_battles)
 
     for player in players:
+        player.reset_battles()
         await player.stop_listening()
 
 
 @pytest.mark.asyncio
 async def test_small_cross_evaluation_gen8():
     await asyncio.wait_for(
-        simple_cross_evaluation(5, format_="gen8randombattle"), timeout=10
+        simple_cross_evaluation(3, format_="gen8randombattle"), timeout=20
```

Comment:
Did you encounter issues regarding execution time with this (and following tests)?

Reply:
The timeout is to counter cases where battles happen to be very long. It didn't happen all that often but I think it's better to be on the safe side with this. This way you know if a test fails, it's not because all the battles took 100 turns.
Regarding the number of battles it's a matter of memory constraint on CircleCI: a medium docker container allows 4 Gb or RAM. The server alone takes up 1.5 Gb and the tests are all called with a single pytest (which brings its own overhead) and, to my knowledge, without garbage collecting the battles. During testing i noticed a lot of times CircleCI killed the build because of RAM constraints.